public/static/plugins/perf/js/trend_chart.js

```diff
@@ -141,31 +161,44 @@ var drawSingleTrendChart = function(params) {
       height: cfg.container.height,
     })
 
+  var colors = d3.scale.category10();
+  // FIXME Force color range 'initialization'. Might be a bug in d3 3.5.3
+  // For some reason, d3 will not give you, let's say, the third color
+  // unless you requested 0, 1 and 2 before.
+  for (var i = 0; i < cfg.knownLevelsCount; i++) colors(i);
+
   var ops = _.pluck(series, cfg.valueAttr);
   var opsValues = _.pluck(series, 'ops_per_sec_values');
   var avgOpsPerSec = d3.mean(ops)
 
   var allLevels = _.pluck(series[0].threadResults, 'threadLevel')
-  var levels = threadMode == MAXONLY
-    ? [_.max(allLevels, function(d) { return +d })]
-    : allLevels
-
-  var levelsMeta = _.map(levels, function(d, i) {
-    var data = {
-      name: d,
-      idx: _.indexOf(allLevels, d),
-    };
-
-    var match = cfg.knownLevels[d];
-
-    data.colorId = match
-      ? match.colorId
-      : cfg.knownLevelsCount + i
-    return data
-  })
+
+  var levelsMeta = threadMode == MAXONLY
+    ? [{name: 'MAX', idx: cfg.valueAttr, color: '#484848'}]
```

Comment:
Could we have the max only default be the first color? Or something different than gray - I feel like this is a bit hard on the eyes

Reply:
Hi John

From the Perf team, we disagree with this request. If you want to tweak the shade of gray that's fine. But it should not be set to the first color. The first color - as currently implemented - is the lowest thread level. In many cases it will be the very opposite of the max timeseries. It would be confusing for those two timeseries to have the same color.

Basically, the way this works now, in this patch, the Max only color needs to be unique.

Reply:
I see - in that case we can disregard this comment. This looks good to me but I'll wait for Henrik to approve as well before merging